### DISCUSSION TITLE
Speclib generation from custom metagenome data

### DISCUSSION THREAD

**Discussion by exploradora623** [vdemichev/DiaNN#1301] [Q&A]
Hello! I have a .FASTA generated by our collaborators who did the metagenome sequencing for my metaproteomics project. I'd like to use DIA-NN to generate a spectral library, but I keep getting "length warnings" with my FASTA checker and DIA-NN crashed after a few days of attempting it. The format is a bit strange compared to the reference proteomes I am used to working with, but I even tried re-formatting the amino acid length, removing duplicates, removing the description entirely and replacing with "protein X" and it didn't resolve the issue. Has anyone encountered this or know what is wrong with the format?  Thank you very much!

Original: 
>L_E1_T20_B_bin.12.renamed_L_E1_T20_B_bin.12_scaffold_72_2 rank: A; chrs:EAG08_20015 (K11753) ribF; riboflavin kinase / FMN adenylyltransferase [EC:2.7.1.26 2.7.7.2] (db=kegg)
MVNCYIDTLLHYTYICCQNFSVLNIFRNFNDYHSEKPLALSLGMFDGVHLGHKYIIDELKKVGSENDLETAILTFWPHPRFVFNPNEDLKLLNTIEEKKFLMEKYNINNLFVKEFDDEFRNLTGEEFVRQILIAKLNVKYLIIGYDHSFGKNKSGNFELLQKLSKELDFEVEQMEAINIHENNISSTKVRNALLAGNIKDANEMLGYSYSVSGMVVHGKKLGRTIGYPTANIETESIKLLPKKGAYIVEVEVNDQQYKGMLSIGTNPTVNGEKLTVEVYILDFNEDIYDKKITVKFRDFLHEEIKFEGLEKLVERLDEDKRITENFAF*

Modified: 
>protein_0000001 rank: A; chrs:EAG08_20020 (K02112) ATPF1B, atpD; F-type H+/Na+-transporting ATPase subunit beta [EC:7.1.2.2 7.2.2.1] (db=kegg); Duplicate proteins: L_E1_T20_C_bin.12.renamed_L_E1_T20_C_bin.12_scaffold_44_409 
MIKIFFLAIAGLSRFIIFAPRKNLAMANQIKGKISQIIGPVIDVVFNNVEAIPSIYDALE
ITKGNGEKVVLEVEQHIGEDTVRCIAMDATDGLQRGQEVIGYGNPIMMPIGEAVNGRLFN
VVGDAIDGLQNISKEGGLPIHRPAPKFDQLSTSAEVLFTGIKVIDLVEPYAKGGKIGLFG
GAGVGKTVLIQELINNIAKGHGGLSVFAGVGERTREGNDLLREMLESGIIKYGDDFMHSM
ENGGWDLSKVDLEVMKESKAAFVFGQMNEPPGARARVALSGLTLAEYYRDGGESGQGRDV
LFFVDNIFRFTQAGSEVSALLGRMPSAVGYQPTLASEMGAMQERITSTKNGSITSVQAVY
VPADDLTDPAPATTFAHLDATTVLDRKIASLGIYPAVDPLASTSRILAPEIIGEEHYNCA
QRVKEILQRYKALQDIIAILGMEELSEEDKSVVYRARKVQRFLSQPFHVAEQFTGIPGSL
VDIKDTIKGFNMIMDGELDHLPEAAFNLKGTIEEAIEAGQKMLADNA

**Comment by vdemichev** [comment]
Hi,

I would in general recommend to just format uniprot-like. At the minimum, you just need the protein sequence ID, and it will work fine in DIA-NN. That is format like ">tr|sequence_id_here" 

Best,
Vadim
